parsers/src/main/scala/uk/gov/ons/addressIndex/parsers/Tokens.scala

```diff
@@ -55,11 +56,43 @@ object Tokens extends CrfTokenable {
   override def normalise(tokens: Array[String]): Array[String] = tokens.map(_.toUpperCase)
 
   def tokensToMap(tokens: Seq[CrfTokenResult]): Map[String, String] = {
-    tokens.groupBy(_.label).map {
+    val validatedTokens = validatePostCode(tokens.groupBy(_.label))
```

Comment:
The aim of the function `tokensToMap` is to transform a list of tokens into a `Map` with a label as a key and concatenated value as a value, why do we have a validation (of the postcode) inside of it?